src/fundamental/macro-fundamental.h

```diff
@@ -249,6 +249,30 @@
                         CONST_ISPOWEROF2(_x);                          \
                 }))
 
+#define ADD_SAFE(ret, a, b) (!__builtin_add_overflow(a, b, ret))
+#define INC_SAFE(a, b) __INC_SAFE(UNIQ, a, b)
+#define __INC_SAFE(q, a, b)                                     \
+        ({                                                      \
+                const typeof(a) UNIQ_T(A, q) = (a);             \
```

Comment:
Hmm, why `const`? We actually want to change the value

Reply:
The const refers to the pointer, not the value. So `A` is a constant pointer to some int-like type. It's not a pointer to a constant int-like type.

Usually the `const` put out front like this applies to the value being pointed at (i.e. `const int * A = &a` is a pointer to a constant int), but the `typeof` adds a level of indirection: now `A` is a constant `typeof(a)` (special syntax for putting `const` out front no longer applies!), where `a` is a pointer to an int, so `A` is a constant pointer to an int (not a pointer to a constant int). This is what we want. 

`const` is always confusing for me in C. Gotta look it up every time :tired_face:

Reply:
Ugh, right. Ignore me then (not a `typeof` guru ;-)